pyquil/quilbase.py

```diff
@@ -28,6 +28,10 @@
                              format_parameter, unpack_qubit)
 
 
+if TYPE_CHECKING:
+    from pyquil.paulis import PauliSum
```

Comment:
what does this do?

Reply:
magic. see previous comment [here](https://github.com/rigetti/pyquil/pull/1125#discussion_r358810527) and/or mypy docs [here](https://mypy.readthedocs.io/en/latest/common_issues.html#import-cycles).

tl;dr it's an escape hatch provided by mypy to allow you to import a module when you need the types only for type checking purposes, but importing the module at runtime would result in an import cycle.